### DISCUSSION TITLE
Receiving "Request failed with status code 502" on Dashboard, Alerts, Hunt, Cases Screens

### DISCUSSION THREAD

**Discussion by postrow** [Security-Onion-Solutions/securityonion#11500] [2.4]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

other (please provide detail below)

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

16GB

### Storage for /

250GB

### Storage for /nsm

1,015.75 GB / 558.69 GB Free

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues

### Detail

Base install of SO v2.4.10 (Hotfix: 20230821) in a Nested ESXi lab environment using VMware port mirroring.  
After a week or so of running SO in our lab environment  we started receiving "502 Error" messages on the aforementioned pages. This info was pulled from logs within "/opt/so/log/elasticsearch".

[2023-10-03T23:59:07,620][WARN ][org.elasticsearch.cluster.routing.allocation.DiskThresholdMonitor] high disk watermark [85%] exceeded on [J9Dl1_55RK2BWQAEMQL3zw][securityonion][/usr/share/elasticsearch/data] free: 17.3gb[11%], shards will be relocated away from this node; currently relocating away shards totalling [0] bytes; the node is expected to continue to exceed the high disk watermark when these relocations are complete

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/ead2fc23-c491-4350-b232-fb70fa942254)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/fd0b56fb-3e20-4246-bb01-5f34cede3d95)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/7703f12f-7b55-4b2b-b1fe-620d6d409dea)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/c20f4497-591e-4e2a-b709-3c5b0251b0fd)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/8011d564-dc84-4032-83c9-252a5a0e2763)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/28dd843d-df09-49e8-8262-cafae3f4f3d7)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by Mav1814** [comment]
Hello, 
It can be about disk space. 
Check it .

**Comment by postrow** [comment]
Thanks for the suggestion - NSM is at 89%

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/589aec1b-057e-4b59-baa9-285000d3a9cc)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/e7d0c8e4-fa1a-4cd3-b5ec-a268d0f7c3e4)

**Reply by Mav1814** [reply]
It can be because of the disk space in nsm. Try to add a new disk. 
The documentation is here 
[https://docs.securityonion.net/en/2.4/new-disk.html?highlight=adding%20new%20disk](url)

**Reply by postrow** [reply]
Mav1814 - Thanks again for the suggestion.  I'm very happy to report that after we expanded the LVM hard disk drive volume from 250G to 500GB, the 502 issues disappeared.  What I'm concerned with is making sure this issue doesn't occur again,  Since our SO v2.4.10 instance in running in a (non-production) test lab environment, I can't imagine that we would have required anything more than the minimum specs.  Is it possible to adjust/tune the index data retention policy to a lower value. Say, maybe 30 days or less?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/170d10aa-73b5-433a-9126-5a4f6e246882)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/70fb6198-437f-4d0a-910b-ac57e80836af)

**Reply by dougburks** [reply]
2.4.10 had an issue where steno could use too much disk space allowing Elasticsearch to hit the watermark. This has been improved in 2.4.20:
https://github.com/Security-Onion-Solutions/securityonion/issues/11305

2.4.20 was released last Friday:
https://blog.securityonion.net/2023/10/security-onion-2420-now-available.html

and a hotfix was released yesterday:
https://blog.securityonion.net/2023/10/security-onion-2420-hotfix-20231012-now.html

We recommend updating as there were many other fixes:
https://docs.securityonion.net/en/2.4/release-notes.html

You can learn more about updating via soup in our documentation:
https://docs.securityonion.net/en/2.4/soup.html

**Reply by postrow** [reply]
Hello Doug -Thank you for taking the time to respond to my post.  I will definitely update to the latest release/hotfix.  

-Best Regards,

**Reply by postrow** [reply]
I updated Security Onion to 2.4.20, and unfortunately, after a reboot, I'm receiving a search query failure. I performed this upgrade on a different instance of Security Onion v2.4.10, and not on the instance I expanded the volume disk space on yesterday. NSM on this updated v2.4.20 image is reporting 89% usage.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/0179b8a6-67db-491b-bed7-d7dec5a71d10)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/d271c542-fe52-4a9a-85cd-926f0707fc67)

[SO Logs_2.4.20.zip](https://github.com/Security-Onion-Solutions/securityonion/files/12895694/SO.Logs_2.4.20.zip)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/39a2a225-3435-423e-bbef-af23aa4550de)

**Reply by postrow** [reply]
I extend the LVM Logical Volume on the updated SO v2.4.20 image and the search query failure is no longer appearing.
	•	sudo lvextend -l +100%FREE /dev/system/nsm
	•	sudo xfs_growfs /dev/system/nsm


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/18b1bc8f-c83f-4253-b995-ee356703c516)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/f1d10283-f39f-423a-8c5d-ed87d0f0202e)

**Reply by seyiakilude** [reply]
Hi Everyone

I am having issues with adding beats agent to Elasticsearch on Kibaba with the following errors. I have been struggling with it for days
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/728f2015-6161-4c15-9cff-18af0571d171)
![WhatsApp Image 2023-11-08 at 23 16 02](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/ea8dc853-e27b-4e50-9453-8adef77fcce8)

**Reply by seyiakilude** [reply]
![WhatsApp Image 2023-11-08 at 23 15 38](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/94cb250f-c965-44dc-9ea9-09bb4d993314)
![WhatsApp Image 2023-11-08 at 23 17 37](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/a2dca0f6-e6eb-4fcd-bede-a6137254a158)
![WhatsApp Image 2023-11-08 at 23 16 49](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/6c8fba3e-b403-4bac-ae93-433d0d674ff0)
![WhatsApp Image 2023-11-08 at 23 16 02](https://github.com/Security-Onion-Solutions/securityonion/assets/25613756/752535cb-09cb-4a90-a165-f0be04b31f3b)

**Reply by seyiakilude** [reply]
1. I tried modifying the hostgroup firewall setting on securityonion but still not working
2. I cant ping the securityonion server but I can access the Securityonion from the web console
3. I am trying to import netflow data into ELK on Securityonion but I keep receiving connection attempts failed on both port 8220 and 9200

**Reply by dougburks** [reply]
@seyiakilude Instead of hijacking somebody else's discussion, please start your own discussion at https://securityonion.net/discuss

**Reply by seyiakilude** [reply]
@dougburks, I am referring to the same server as I am part of the team working with this server. However, I have created another discussion. Thanks

**Comment by ponte19** [comment]
Same Problem for me, /nsm also nearly full and error 502. What is the solution?

**Reply by dougburks** [reply]
@ponte19 From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.
